extensions/2.0/Vendor/MSFT_CollisionPrimitives/README.md

```diff
@@ -0,0 +1,75 @@
+
+To describe the shape that represents this node, should use, colliders must define exactly one of the following properties:
+
+| |Type|Description|
+|-|-|-|
+|**sphere**|`object`|A sphere centered at the origin in local space.|
+|**box**|`object`|An axis-aligned box centered at the origin in local space.|
+|**capsule**|`object`|A capsule centered at the origin and aligned along the Y axis in local space.|
+|**cylinder**|`object`|A cylinder centered at the origin and aligned along the Y axis in local space.|
+|**convex**|`object`|A convex hull wrapping a `mesh` object.|
+|**trimesh**|`object`|A triangulated representation of a `mesh` object.|
+
+The sphere, box, capsule, cylinder and convex types should all produce _solid_ colliders by default - a rigid body anywhere inside the shape should be pushed out of it.
+However the trimesh type always represents an infinitely thin shell or sheet - for example a trimesh collider created from a `mesh` object in the shape of a box will be represented as a hollow box.
+
+If you want your collider to have an offset from the local space (for example a sphere _not_ centered at local origin, or a rotated box), you should add an extra node to the hierarchy and apply your transform and your collider properties to that.
+
+**Collision Filtering**
+
+By default each `collider` can generate collisions with every other `collider`, provided they are sufficiently close. If you want certain objects in your scene to ignore collisions with others, you can provide the following optional collider properties:
+
+| |Type|Description|
+|-|-|-|
+|**collisionSystems**|`[string]`|An array of arbitrary strings indicating the `system` a node is in.|
+|**notCollideWithSystems**|`[string]`|An array of strings representing the systems which this node will _not_ collide with|
+|**collideWithSystems**|`[string]`|An array of strings representing the systems which this node can collide with|
+
+Both `collideWithSystems` and `notCollideWithSystems` are provided so that users can override the default collision behaviour with minimal configuration -- only one of these should be specified per-collider. Note, given knowledge of all the systems in a scene and one of the vvalues `notCollideWithSystems`/`collideWithSystems` the unspecified field can be calculated: `collideWithSystems = notCollideWithSystems'`
+
+`notCollideWithSystems` is useful for an object which should collide with everything except those listed in `notCollideWithSystems` (i.e., used to opt-out of collisions) while `collideWithSystems` is the inverse -- the collider should _not_ collide with any other collider excpet those listed in `collideWithSystems`
+
+A node `A` will collide with node `B` if `A.collisionSystem ⊆ B.collideWithSystems && A.collisionSystem ⊄ B.notCollideWithSystems`
```

Comment:
Are collisions scoped to the glTF? In other words, with this specification alone, can two separate glTFs produce collision events? There's [a concern](https://github.com/KhronosGroup/glTF/pull/2258#issuecomment-1436120156) the number of unique systems specified may exceed a reasonable count and this problem would be exacerbated in the multi-glTF case.

Reply:
No, they shouldn't be scoped to a single file; applications load assets from multiple files and developers would expect that it should be possible to control whether two objects collide even if those objects come from different files.

An earlier version of this spec did specify the filter using a bitfield but I found that too limiting. It's not easy for non-technical artists to understand and it's not portable across different asset sources. I really want to enable the use-case where you can download an asset from an arbitrary source and at least have _some_ confidence that the filtering will work correctly (because with this strategy, you can use strings to semantically label your intent, versus a bitfield where -say- bit 5 by itself is meaningless).

For what it's worth, we had this same discussion in the USD Physics working group and came to a very similar solution (there are some differences due to the way USD allows scene composition via layers, but the core idea is the same.)

Reply:
Sounds good. I completely agree with strings over bitfields.

Reply:
What should the behavior be defined as if 5 glTF files each specify 20 different collision filtering system strings, and it cannot be represented in engines as bitwise anymore? Do we use bits until there are no more, and then ignore the rest of the collision filtering system strings for all future file imports?

Reply:
The edge case I mentioned above still needs to be figured out. Is it undefined behavior?